vortex-expr/src/transform/remove_merge.rs

```diff
@@ -0,0 +1,111 @@
+use vortex_dtype::DType;
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::traversal::{MutNodeVisitor, Node, TransformResult};
+use crate::{get_item, pack, ExprRef, Merge, VortexExpr};
+
+/// Replaces [Merge] with combination of [GetItem] and [Pack] expressions.
+pub fn remove_merge(e: ExprRef, scope_dt: &DType) -> VortexResult<ExprRef> {
+    let mut transform = RemoveMergeTransform {
+        scope_dtype: scope_dt,
+    };
+    e.transform(&mut transform).map(|e| e.result)
+}
+
+struct RemoveMergeTransform<'a> {
+    scope_dtype: &'a DType,
+}
+
+impl MutNodeVisitor for RemoveMergeTransform<'_> {
+    type NodeTy = ExprRef;
+
+    fn visit_up(&mut self, node: ExprRef) -> VortexResult<TransformResult<Self::NodeTy>> {
+        if let Some(merge) = node.as_any().downcast_ref::<Merge>() {
```

Comment:
You should make a validity/nullability assertion in this visitor

Reply:
Done